core/schema/src/main/liquibase/21.0.0/changelog.xml

```diff
@@ -57,4 +57,54 @@
       );
     </sql>
   </changeSet>
+  <changeSet id="21.0.0-event-parameters" author="fooker">
+    <createTable tableName="event_parameters">
+      <column name="eventid" type="INTEGER">
+        <constraints nullable="false"/>
+      </column>
+      <column name="name" type="VARCHAR(256)">
+        <constraints nullable="false"/>
+      </column>
+      <column name="value" type="TEXT">
+        <constraints nullable="false"/>
+      </column>
+      <column name="type" type="VARCHAR(256)">
+        <constraints nullable="false"/>
+      </column>
+    </createTable>
+    <addPrimaryKey tableName="event_parameters"
+                   columnNames="eventid,name"
+                   constraintName="pk_eventparameters" />
+    <addForeignKeyConstraint baseTableName="event_parameters" baseColumnNames="eventid"
+                             constraintName="fk_eventparameterseventid"
+                             referencedTableName="events"
+                             referencedColumnNames="eventid"
+                             onDelete="CASCADE"/>
+
+    <dropColumn tableName="alarms" columnName="eventparms" />
+    <addColumn tableName="alarms">
```

Comment:
After discussing this with @pbrane, we think we should just always use `lasteventid` as the source of the parameters rather than introducing another column and constraint.